parsesftpdump.py

```diff
@@ -7,30 +7,65 @@
+UTC_DATESTAMP_FSTR = '%Y-%m-%dT%H:%M:%SZ'
+MARCFILE_PREFIX = 'alma_bibs__'
+MARCFILE_EXTENSION = 'xml'
+ingest_command = Variable.get("AIRFLOW_HOME") + "/dags/cob_datapipeline/scripts/ingest_marc_multi.sh"
+marcfilepath = Variable.get("AIRFLOW_DATA_DIR") + "/sftpdump/"
+
+# parse filename for datestamp
+# e.g. alma_bibs__2018041306_6863237930003811_new_61.xml
+def get_dump_date(marcdircontents):
+    datestamp = None
+    datestampregex = re.compile(r"alma_bibs__(.*)_(.*).xml")
+    marcfileregex = re.compile(MARCFILE_PREFIX + r".*\." + MARCFILE_EXTENSION + "$")
+    for infilename in marcdircontents:
+        if marcfileregex.search(infilename):
+            rxresults = datestampregex.search(infilename)
+            if rxresults is not None:
+                datestamp = rxresults.group(1)
+                break
+    return datestamp
+
+def parse_sftpdump_dates(ds, **kwargs):
     date = None
     if os.path.isfile(ingest_command):
         marcdircontents = os.listdir(marcfilepath)
-        marcfileregex = re.compile(marcfile_prefix + ".*\." + marcfile_extension + "$")
-        #parse filename for datestamp e.g. alma_bibs__2018041306_6863237930003811_new_61.xml
-        datestampregex = re.compile(r"alma_bibs__(.*)_(.*).xml")
-        for infilename in marcdircontents:
-            if marcfileregex.search(infilename):
-                m = datestampregex.search(infilename)
-                if m is not None:
-                    datestamp = m.group(1)
-                    #  MAYBE TRY TO PARSE THESE OUT IN THE FUTURE?
-                    Variable.set("almaoai_last_num_oai_update_recs", 0)
-                    Variable.set("almaoai_last_num_oai_delete_recs", 0)
-                    Variable.set("almaoai_last_harvest_date", 0)
-                    break
-        date = time.strptime(datestamp[:8], "%Y%m%d")
-        Variable.set("almaoai_last_harvest_date", time.strftime(UTC_DATESTAMP_FSTR, date))
-        Variable.set("almaoai_last_harvest_from_date", time.strftime(UTC_DATESTAMP_FSTR, date))
+        datestamp = get_dump_date(marcdircontents)
+        if datestamp is not None:
+            #  MAYBE TRY TO PARSE THESE OUT IN THE FUTURE?
+            Variable.set("almaoai_last_num_oai_update_recs", 0)
+            Variable.set("almaoai_last_num_oai_delete_recs", 0)
+            date = time.strptime(datestamp[:8], "%Y%m%d")
+            Variable.set("almaoai_last_harvest_date", time.strftime(UTC_DATESTAMP_FSTR, date))
+            Variable.set("almaoai_last_harvest_from_date", time.strftime(UTC_DATESTAMP_FSTR, date))
+        else:
+            raise Exception("Cannot find a datestamp in {}".format(marcfilepath))
+    else:
+        raise Exception(str(os.path.isfile(ingest_command)) +
+                        " Cannot locate {}".format(ingest_command))
+
+def renamesftpfiles_onsuccess(ds, **kwargs):
+    marcdircontents = os.listdir(marcfilepath)
+    datestamp = get_dump_date(marcdircontents)
```

Comment:
do we want to pass the datestamp used in processing functions above in here instead of re-calculating? Edge case that another export occurs while a full reindex is occurring.

Reply:
This is occurring on the local filesystem, not the sftp server. We don't allow two reindex jobs to run concurrently, so there can only ever be one dump date present.